src/filter/pixs0r/pixs0r.cc

```diff
@@ -0,0 +1,218 @@
+#include <random>
+
+struct pixshift0r
+{
+    unsigned int m_width;
+    unsigned int m_height;
+
+    unsigned int m_shift_intensity;
+    unsigned int m_block_height;
+    
+    // If m_block_height == 0, then these bound block heights
+    unsigned int m_block_height_min;
+    unsigned int m_block_height_max;
+
+    // FIXME: It might be better to make it global variable.
+    std::random_device m_rng_dev;
+
+    std::uniform_int_distribution<long long int> m_shift_rng;
+    std::uniform_int_distribution<unsigned int> m_block_height_rng;
+
+    pixshift0r(unsigned int width, unsigned int height)
+        : m_width(width), m_height(height), m_block_height(0) {}
+
+    void process(const uint32_t *inframe, uint32_t *outframe)
+    {
+        for (unsigned int b = 0; b < m_height;)
+        {
+            unsigned int block_height = m_block_height ? m_block_height : m_block_height_rng(m_rng_dev);
+
+            // Number of rows to shift is either block size or
+            // what we can *safely* operate on (avoids corruption).
+            block_height = std::min(block_height, m_height - b);
+
+            long long shift = m_shift_rng(m_rng_dev);
+
+            for (unsigned int j = 0; j < block_height; ++j)
+            {
+				// NOTE: Faulty implementations, such as FFmpeg don't
+				// check for (width % 8 == 0 && height % 8 == 0).
```

Comment:
This is no longer strictly enforced and should not be required in new filters. It is very rare that a frei0r actually requires this. Do you know of any? This rule came from a long time ago when frei0r was used primarily in VJ applications applied to whole frames of video for performance reasons. The fact most filters lack optimization or multithreading really limited this performance rationale while video was SD. For the past decade the most common usage of frei0r is in video editors where you may be working with frames scaled or cropped to any size because either they will be composited or because modern codecs do not require multiples of 8. But no one has been bold enough to remove it from the "spec" because there is not much of a project community or active leadership.

Reply:
I see. It seemed very weird in fact but FFmpeg seems to adhere to that standard (errnously of course).